### PR TITLE
feat: Add verify and expiry methods to AppIntegrityManager

### PR DESCRIPTION
**Pull request by alex-bradbury-gds**
- Add method to verify a given attestation matches the local public key
- Add method to extract expiry time from attestation
- Rename `ClientAttestationManager` to `AppIntegrityManager`, given the PoP is not really to do with the attestation

### CODE REVIEW

File: app/src/main/java/uk/gov/android/authentication/integrity/FirebaseAppIntegrityManager.kt [hunk 1]
```diff
@@ -62,6 +65,32 @@ class FirebaseClientAttestationManager(
         }
     }
 
+    override fun verifyAttestationJwk(attestation: String): Boolean {
```
Comment (ThomasIent) [inline app/src/main/java/uk/gov/android/authentication/integrity/FirebaseAppIntegrityManager.kt:68]: I  think that it isn't particularly object orientated to be passing things around as string and converting them to objects only when we need to do object comparison, but for the sake of expediency

File: app/src/main/java/uk/gov/android/authentication/integrity/keymanager/ECKeyManager.kt [hunk 2]
```diff
@@ -56,8 +57,10 @@ class ECKeyManager : KeyStoreManager {
             update(input)
             sign()
         }
-        val verifyResult = verify(input, signature)
-        Log.d("VerifySignature", "$verifyResult")
+        if (BuildConfig.DEBUG) {
```
Comment (ThomasIent) [inline app/src/main/java/uk/gov/android/authentication/integrity/keymanager/ECKeyManager.kt:60]: I had to wrap the Log.d above in a `if (BuildConfig.DEBUG)` weird that the linter only complains about this one for you.

**Review by ThomasIent — APPROVED** [review]
This didn't build for me, but if it is working then so be it.

File: app/src/main/java/uk/gov/android/authentication/integrity/FirebaseAppIntegrityManager.kt (hunk 1 above)
Comment (alex-bradbury-gds) [reply app/src/main/java/uk/gov/android/authentication/integrity/FirebaseAppIntegrityManager.kt:68]: generally agree, but as we discussed, the consumer has no need to do anything with the attestation, so seems overkill to deserialize etc.

File: app/src/main/java/uk/gov/android/authentication/integrity/keymanager/ECKeyManager.kt (hunk 2 above)
Comment (alex-bradbury-gds) [reply app/src/main/java/uk/gov/android/authentication/integrity/keymanager/ECKeyManager.kt:60]: Yeah, weird this. This particular section will likely be removed soon anyway

**Pull request merged**